### ISSUE TITLE
nearly impossible to release patch versions once we update CI versions

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#5601]
We updated to Go 1.13 in our CI in https://github.com/sourcegraph/sourcegraph/pull/5485 which makes it nearly impossible to cut a patch release on older 3.6/3.7 branches cut just a month/two months ago.

Basically, we have to backport Go 1.13 support to those versions in order to cut a patch release. Can we avoid this somehow?

**Comment by slimsag** [comment]
We discussed this at the Distribution sync. Takeaways:

- The Go version was the primary symptom here, but the underlying cause can affect any of our tooling dependencies (nodejs version, Go version, etc).
- We concluded having one version in CI is not sufficient, and supporting all versions would be non-trivial.


We concluded the next step here is to take the Go and NodeJS dependency installation and install those as part of our CI pipeline in this repository -- not in our base buildkite image as they are today. We will need to make use of a cache in buildkite to ensure we don't suffer the overhead of installing each dependency. There appear to be multiple caches we could use for this, with one being https://github.com/danthorpe/cache-buildkite-plugin
